src/Raven.Server/Documents/Handlers/Processors/Indexes/IndexHandlerProcessorForConvertAutoIndex.cs

```diff
@@ -0,0 +1,716 @@
+
+        static Dictionary<string, IndexFieldOptions> ConstructFields(AutoIndexDefinition autoIndex, AutoIndexConversionContext context)
+        {
+            Dictionary<string, IndexFieldOptions> fields = null;
+            if (autoIndex.MapFields is { Count: > 0 })
+            {
+                foreach (var kvp in autoIndex.MapFields)
+                {
+                    var fieldNames = GenerateFieldName(kvp.Key, kvp.Value.Indexing);
+
+                    foreach (var f in fieldNames)
+                    {
+                        HandleFieldIndexing(f.FieldName, f.Indexing);
+                        HandleSpatial(f.FieldName, kvp.Value.Spatial, context);
+                        HandleStorage(f.FieldName, kvp.Value.Storage);
+                        HandleSuggestions(f.FieldName, kvp.Value.Suggestions);
+                    }
+                }
+            }
+
+            return fields;
+
+            IndexFieldOptions GetFieldOptions(string fieldName)
+            {
+                fields ??= new Dictionary<string, IndexFieldOptions>();
+                if (fields.TryGetValue(fieldName, out var value) == false)
+                    fields[fieldName] = value = new IndexFieldOptions();
+
+                return value;
+            }
+
+            void HandleFieldIndexing(string fieldName, AutoFieldIndexing? fieldIndexing)
+            {
+                if (fieldIndexing.HasValue == false)
+                    return;
+
+                if (fieldIndexing.Value.HasFlag(AutoFieldIndexing.Search))
+                {
+                    var options = GetFieldOptions(fieldName);
+                    options.Indexing = FieldIndexing.Search;
+                    return;
+                }
+
+                if (fieldIndexing.Value.HasFlag(AutoFieldIndexing.Exact))
+                {
+                    var options = GetFieldOptions(fieldName);
+                    options.Indexing = FieldIndexing.Exact;
+                    return;
+                }
+
+                if (fieldIndexing.Value.HasFlag(AutoFieldIndexing.Highlighting))
+                {
+                    var options = GetFieldOptions(fieldName);
+                    options.Indexing = FieldIndexing.Search;
+                    options.Storage = FieldStorage.Yes;
+                    options.TermVector = FieldTermVector.WithPositionsAndOffsets;
+                    return;
+                }
+            }
+
+            void HandleStorage(string fieldName, FieldStorage? fieldStorage)
+            {
+                if (fieldStorage.HasValue == false || fieldStorage == FieldStorage.No)
+                    return;
+
+                var options = GetFieldOptions(fieldName);
+
+                switch (fieldStorage.Value)
+                {
+                    case FieldStorage.Yes:
+                        options.Storage = FieldStorage.Yes;
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
```

Comment:
Shouldn't we handle `FieldStorage.No` ? Even if it's a no op

Reply:
And please provide message to the exception

Reply:
No, this generates a lot of noise, for each field in the definition then it is set to 'No', same for AbstractIndexCreationTask, you are getting `fields x Store(name, No)`